_posts/2024-04-05-Configurable-Automation-for-OpenSearch-ML-Use-Cases.md

```diff
@@ -0,0 +1,133 @@
+
+## Before the Flow Framework plugin
+
+Previously, setting up semantic search involved the steps outlined in the [semantic search documentation](https://opensearch.org/docs/latest/search-plugins/semantic-search/):
+
+1. Create a connector for a remote model, specifying pre- and post-processing functions.
+1. Register an embedding model using the connector ID obtained in the previous step.
+1. Configure an ingest pipeline to generate vector embeddings using the model ID of the registered model.
+1. Create a k-NN index and add the pipeline created in the previous step.
+
+This complex setup typically required you to be familiar with the OpenSearch ML Commons APIs. However, we are simplifying this experience through the Flow Framework plugin. Let's demonstrate how the plugin simplifies this process using the preceding semantic search example.
+
+## With the Flow Framework plugin
+
+In this example, you will configure the `semantic_search_with_cohere_embedding_query_enricher` workflow template. The workflow created using this template performs the following configuration steps:
+
+* Deploys an externally hosted Cohere model
+* Creates an ingest pipeline using the model
+* Creates a sample k-NN index and configures a search pipeline to define the default model ID for that index
+
+### Step 1: Create and provision the workflow
+
+Using the `semantic_search_with_cohere_embedding_query_enricher` workflow template, you provision the workflow with just one required field---the API key for the Cohere Embed model:
+
+```json
+POST /_plugins/_flow_framework/workflow?use_case=semantic_search_with_cohere_embedding_query_enricher&provision=true
+{
+    "create_connector.credential.key" : "<YOUR API KEY>"
+}
+```
+
+OpenSearch responds with a unique workflow ID, simplifying the tracking and management of the setup process:
+
+```json
+{
+  "workflow_id" : "8xL8bowB8y25Tqfenm50"
+}
+```
+
+Note: The workflow in the previous step creates a default k-NN index. The default index name is `my-nlp-index`:
+
+```json
+{
+  "create_index.name": "my-nlp-index"
+}
+```
+
+You can customize the template default values by providing the new values in the request body. For a comprehensive list of default parameter values for this workflow template, see [Cohere Embed semantic search defaults](https://github.com/opensearch-project/flow-framework/blob/2.13/src/main/resources/defaults/cohere-embedding-semantic-search-defaults.json).
+
+### Step 2: Ingest documents into the index
+
+Once the workflow is provisioned, you can ingest documents into the index created by the workflow:
+
+```json
+PUT /my-nlp-index/_doc/1
```

Comment:
We should probably use the path without the document ID included.  Having the `1` here and the `s1` in the "id" field is confusing.  (We can leave out the "id" field as well since it's not the same as the "_id".)